### PR TITLE
FIPS fixes; sequence config; restore old keystore check; link node fo…

### PR DESCRIPTION
**Pull request by david-oc-miller**
…lders

### CODE REVIEW

File: vagrant/provisioning/roles/alfresco/tasks/main.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@
       - libreoffice
       - ImageMagick
       - ImageMagick-devel
+      - python2-jmespath
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/alfresco/tasks/main.yml:9]: this package needed on RHEL 7 for some reason

File: vagrant/provisioning/roles/arkcase-app/tasks/main.yml [hunk 2]
```diff
@@ -911,6 +911,8 @@
       replace: PKCS12
     - regexp: "#\\{systemProperties\\['user.home'\\]\\}/.arkcase/acm/private/encryption-keystore"
       replace: "{{ java_key_store }}"
+    - regexp: "#\\{systemProperties\\['user.home'\\]\\}/.arkcase/acm/private/keystore.old"
+      replace: "{{ java_key_store }}"
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-app/tasks/main.yml:915]: this line lets me install releases 3.3.3 and below

File: vagrant/provisioning/roles/arkcase-post-upgrade/tasks/main.yml [hunk 3]
```diff
@@ -1,7 +1,10 @@
-- name: restore arkcase-runtime.yml
+- name: restore config files 
   become: yes
   become_user: arkcase
-  command: cp {{ config_backup_folder_path }}/acm/acm-config-server-repo/arkcase-runtime.yaml {{ root_folder }}/data/arkcase-home/.arkcase/acm/acm-config-server-repo/arkcase-runtime.yaml
+  command: cp {{ config_backup_folder_path }}/acm/{{ item }} {{ root_folder }}/data/arkcase-home/.arkcase/acm/{{ item }}
+  loop:
+    - "acm-config-server-repo/arkcase-runtime.yaml"
+    - "acmSequenceConfiguration.json"
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/arkcase-post-upgrade/tasks/main.yml:7]: stop blowing away the sequence configuration as requested by Matt

File: vagrant/provisioning/roles/pki/tasks/main.yml [hunk 4]
```diff
@@ -19,9 +19,15 @@
 - name: generate root ca private key
   become: yes
   openssl_privatekey:
-    path: "/etc/tls/private/arkcase-root.pem"
+    path: "/etc/tls/private/arkcase-root-passphrase.pem"
```
Comment (david-oc-miller) [inline vagrant/provisioning/roles/pki/tasks/main.yml:22]: these changes fix PKI generation in fips mode.

Comment (david-oc-miller) [reply vagrant/provisioning/roles/pki/tasks/main.yml:22]: ... we also have to make a change in yarn so it doesn't use md5... fips prevents md5

Comment (david-oc-miller) [reply vagrant/provisioning/roles/pki/tasks/main.yml:22]: specifically, this patch is needed for yarn on fips, but I'd rather keep this a manual step for now:

```bash
[david.miller_admin@eeocppappaapp01 ~]$ diff -Naurb /usr/share/yarn/lib/cli.js.original /usr/share/yarn/lib/cli.js
--- /usr/share/yarn/lib/cli.js.original 2020-02-20 23:18:25.444000000 +0000
+++ /usr/share/yarn/lib/cli.js  2020-02-20 23:19:05.765000000 +0000
@@ -43202,7 +43202,7 @@
 var stream = __webpack_require__(35);
 
 function hash(content) {
-  var type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'md5';
+  var type = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : 'sha1';
 
   return crypto.createHash(type).update(content).digest('hex');
 }
```

**Review by bojmil-armedia — APPROVED** [review]

**Pull request merged**